src/items/rqgItem.ts

```diff
@@ -169,4 +169,111 @@ export class RqgItem extends Item {
   public static localizeItemTypeName(itemType: ItemTypeEnum): string {
     return localize("ITEM.Type" + itemType.titleCase());
   }
+
+  /**
+   * Return the highest priority item matching the supplied rqid and lang from the items in the World.  If not
+   * found return the highest priority item matching the supplied rqid and lang from the installed Compendia.
+   * eg CONFIG.Item.documentClass.getItemByRqid("someid", "es")
+   * @param rqid eg "humanoid-left-arm", "throwing-axe", or "spirit-lore"
+   * @param lang default value is "en", eg "en" or "pl"
+   * @returns
+   */
+  public static async getItemByRqid(
+    rqid: string,
+    lang: string = "en"
+  ): Promise<RqgItem | undefined> {
+    const worldItem = await this.getItemFromWorldByRqid(rqid, lang);
+
+    if (worldItem !== undefined) {
+      return worldItem;
+    }
+
+    const comendiumItem = await this.getItemFromAllCompendiaByRqid(rqid, lang);
+
+    return comendiumItem;
+  }
+
+  private static async getItemFromWorldByRqid(
+    rqid: string,
+    lang: string = "en"
+  ): Promise<RqgItem | undefined> {
+    const candidates = getGame().items?.contents.filter(
+      (i) => i.data.data.rqid === rqid && i.data.data.rqidlang === lang
+    );
+
+    if (candidates === undefined) {
+      return undefined;
+    }
+
+    if (candidates.length > 0) {
+      let result = candidates.reduce((max, obj) =>
+        max.data.data.rqidpriority > obj.data.data.rqidpriority ? max : obj
+      );
+      
+      // Detect more than one item that could be the match
+      let duplicates = candidates.filter(i => i.data.data.rqidpriority === result.data.data.rqidpriority);
+      if (duplicates.length > 1) {
+        const msg = localize("RQG.Item.RqgItem.Error.MoreThanOneRqidMatchInWorld", {
+          rqid: rqid,
+          rqidlang: lang,
+          rqidpriority: result.data.data.rqidpriority,
+        });
+        ui.notifications?.error(msg);
+        console.log(msg + "  Duplicate items: ", duplicates);
+      }
+      return result as RqgItem;
+    } else {
+      const msg = localize("RQG.Item.RqgItem.Error.ItemNotFoundByRqid", {
+        rqid: rqid,
+        rqidlang: lang,
+      });
+      ui.notifications?.warn(msg);
+      console.log(msg);
+      return undefined;
+    }
+  }
+
+  private static async getItemFromAllCompendiaByRqid(
+    rqid: string,
+    lang: string = "en"
+  ): Promise<RqgItem | undefined> {
+    const candidates: RqgItem[] = [];
+
+    for (const pack of getGame().packs) {
+      if (pack.documentClass.name === "RqgItem") {
+        for (const item of await pack.getDocuments()) {
```

Comment:
So after we find a compendium that has RqgItem in it, we should use .filter?  Or maybe .getDocuments() with a query?  I tried this just now and it wasn't working.  I think I need to commit the other stuff from the review and come back to this to avoid getting confused.

Reply:
This has not been addressed as I either don't understand you or wasn't able to get filter to work on a compendium pack.

Reply:
Yes this could be a separate issue since it works as it is and this is more about optimising it.
created #213 for this.